File: android/app/src/main/kotlin/in/nann/smashhockey/ui/KitSound.kt
package `in`.nann.smashhockey.ui

import `in`.nann.smashhockey.core.generated.SoundCue

/**
 * The UI kit's sounds (spec §8.8, §16): what a component plays when it is pressed, refused, flipped,
 * landed, swooped past or stepped. The kit only names the event; the game's playback layer, set here
 * once, plays it. Silent until then (a kit sketch without sound is a genuine state).
 */
object KitSound {
    var play: ((SoundCue) -> Unit)? = null

    fun press() = play?.invoke(SoundCue.UI_BUTTON_PRESS)
    fun refuse() = play?.invoke(SoundCue.UI_ERROR)
    fun flip() = play?.invoke(SoundCue.UI_DIGIT_FLIP)
    fun land() = play?.invoke(SoundCue.UI_PANEL_POP)
    fun swoop() = play?.invoke(SoundCue.UI_CAMERA_WHOOSH_LONG)
    fun sweep() = play?.invoke(SoundCue.UI_CAMERA_WHOOSH_SHORT)
    fun step() = play?.invoke(SoundCue.UI_SLIDER_TICK)
}

File: ios/Sources/UI/KitSound.swift
import SmashCore

/// The 3D UI kit's sounds (spec §8.8): a press, a nope, a flip, a panel landing, a whoosh, a slider's
/// step. The kit only says what happened; the game plays it (set once, by `Game`). Silent until set.
@MainActor
enum KitSound {
    static var play: ((SoundCue) -> Void)?

    static func press() { play?(.uiButtonPress) }
    static func nope() { play?(.uiError) }
    static func flip() { play?(.uiDigitFlip) }
    static func pop() { play?(.uiPanelPop) }
    static func swoop() { play?(.uiCameraWhooshLong) }
    static func sweep() { play?(.uiCameraWhooshShort) }
    static func step() { play?(.uiSliderTick) }
}
